std/file.d

```diff
@@ -5119,3 +5120,77 @@ string tempDir() @trusted
     myFile.write("hello");
     assert(myFile.readText == "hello");
 }
+
+/**
+* Returns the available space in a file system.
+*
+* On `Windows`, this function uses the result of the Windows API function
+* $(LINK2 https://msdn.microsoft.com/en-us/library/windows/desktop/aa364937(v=vs.85).aspx, $(D GetDiskFreeSpaceExW)).
+*
+* On `POSIX` platforms, it uses the $(LINK2 http://man7.org/linux/man-pages/man3/statvfs.3.html,
+* $(D statvfs)) function.
+*
+* Due to inconsistent behaviour with the `statvfs` `Posix` function,
+* `FreeBSD` does not support `getAvailableDiskSpace`.
+*
+* Params:
+*     path = a valid path to any file in the file system
+*
+* Returns:
+*    ulong which contains the available space in bytes
+*
+* Throws:
+*     $(LREF FileException) in case statvfs (`POSIX`) or GetDiskFreeSpaceExW (`Windows`) fails.
+*/
+ulong getAvailableDiskSpace(string path) @trusted
```

Comment:
Needs to be a template to avoid instantiation by default.
Also note that the other functions accept `const(char)[]` which is different to `immutable(char)[]` which is `string`.

Reply:
Oh and ideally the `@trusted` goes inline to only to the needed lines